src/internal/checks_libevent/HMEventLoopLibEvent.cpp

```diff
@@ -271,7 +271,8 @@ HMEventLoopLibEvent::handleDNSTimeout(evutil_socket_t fd, short what, void* arg)
                 data->m_hostname.c_str());
         currentState->m_dnsCache.queueDNSQuery(data->m_hostname,
                 data->m_dnsHostCheck,
-                state->m_workQueue);
+                state->m_workQueue,
+                state->getStateVersion());
```

Comment:
state doesnn't have function getStateVersion() , did you mean currentState->getStateVersion()